### DISCUSSION TITLE
Union query across all named graphs

### DISCUSSION THREAD

**Discussion by adlerfaulkner** [comunica/comunica#1287] [Q&A] [answered]
Is there a way in which I can use communica, and more importantly any rdf/js query engine, where the default graph is a union of all named graphs? 

Such functionality is mentioned as possible [in the RDF/JS Query Specification](https://rdf.js.org/query-spec/#informative-references:~:text=sometimes%20called%20the%20union%20graph), though I am not sure how to set that option when using [quadstore-comunica](https://www.npmjs.com/package/quadstore-comunica) or [@comunica/query-sparql-rdfjs](https://www.npmjs.com/package/@comunica/query-sparql-rdfjs).

I understand that this is dependent on the SPARQL implementation but I am used to it because that's what AWS Neptune does [https://docs.aws.amazon.com/neptune/latest/userguide/best-practices-sparql.html#best-practices-sparql-query](https://docs.aws.amazon.com/neptune/latest/userguide/best-practices-sparql.html#best-practices-sparql-query)

**Comment by adlerfaulkner** [comment]
Seems like this could be due to SparqlAlgebra.js defaulting to setting the graph to `DefaultGraph` for all triple patterns not wrapped in `GRAPH ?g { ... }` as shown [here](https://github.com/joachimvh/SPARQLAlgebra.js#translate) rather than undefined or null as mentioned in the [RDF/JS Query Specification](https://rdf.js.org/query-spec/#informative-references:~:text=sometimes%20called%20the%20union%20graph)

**Comment by rubensworks** [comment, marked as answer]
This is possible by enabling this flag: https://comunica.dev/docs/query/advanced/context/#17--union-default-graph

**Reply by adlerfaulkner** [reply]
Ah great, thanks! I looked for this for a while! I think from a usability perspective on Communica it would really help to be able to search the documentation. I think I was also thrown off track by the fact that options like this are specified through `context`. I was relating context to JSON-LD context and thus thought it was only for prefixes and baseIRI. Maybe consider renaming it to `options`?
